packages/api/src/router/activity.ts

```diff
@@ -1,22 +1,52 @@
 import { addWeeks } from "date-fns";
 import { z } from "zod";
 
-import { adminProcedure, createTRPCRouter, protectedProcedure } from "../trpc";
+import { adminProcedure, createTRPCRouter, publicProcedure } from "../trpc";
+
+function isValidDate(dateString: string): boolean {
+  const parsedDate = Date.parse(dateString);
+  if (isNaN(parsedDate)) {
+    return false;
+  }
+
+  const date = new Date(parsedDate);
+  const formattedDate = date.toISOString().split("T")[0];
+
+  return dateString === formattedDate;
+}
 
 export const activityRouter = createTRPCRouter({
-  getSchedule: protectedProcedure
-    .input(z.object({ day: z.date() }))
+  getSchedule: publicProcedure
```

Comment:
Thanks for fixing this endpoint! I really appreciate it. Good work, last thing to do is switch this back to a `protectedProcedure` since its only for authenticated users.

Sorry! That's the last thing